algorithms/src/merkle_tree/merkle_tree.rs

```diff
@@ -118,10 +119,10 @@ impl<P: MerkleParameters> MerkleTree<P> {
     }
 
     pub fn rebuild<L: ToBytes, I: ExactSizeIterator<Item = L>, J: ExactSizeIterator<Item = L>>(
-        &mut self,
+        &self,
```

Comment:
would it be possible to consume `self` here instead of just borrowing it?

Reply:
@ljedrz No. We have to borrow it since the caller will only have an `Arc<MerkleTree>`